### DISCUSSION TITLE
More information in json file for ASE interface is needed

### DISCUSSION THREAD

**Discussion by 1041176461** [deepmodeling/abacus-develop#5818] [General]
### Background

For ASE interface, information in json file is not enough. 

### Describe the solution you'd like

Velocity, energy eigenvalues (SCF and NSCF), occupations(SCF and NSCF), k-points, k-weights, dipole, atomic magnetic moment, MD informations (STRU, Temperature, Pressure, Energy, Potential, Kinetic), charge density are needed

### Task list only for developers

- [ ] Notice possible changes of behavior
- [ ] Explain the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi

### Task list for Issue attackers (only for developers)

- [ ] Review and understand the proposed feature and its importance.
- [ ] Research on the existing solutions and relevant research articles/resources.
- [ ] Discuss with the team to evaluate the feasibility of implementing the feature.
- [ ] Create a design document outlining the proposed solution and implementation details.
- [ ] Get feedback from the team on the design document.
- [ ] Develop the feature following the agreed design.
- [ ] Write unit tests and integration tests for the feature.
- [ ] Update the documentation to include the new feature.
- [ ] Perform code review and address any issues.
- [ ] Merge the feature into the main branch.
- [ ] Monitor for any issues or bugs reported by users after the feature is released.
- [ ] Address any issues or bugs reported by users and continuously improve the feature.

**Comment by WHUweiqingzhou** [comment]
@pxlxingliang, could you have a look?

**Comment by QuantumMisaka** [comment]
@pxlxingliang 
Can we divide these above information by 
- Molecular simulation part (T,P and etc)
- Atom properties (STRU, velocity, magnetic moments , etc)
- Electronic structure part (charge density, K-points)
and format them in json file?

Also @1041176461 I think the different part of ASE-ABACUS interface needs different information, can we list and organize them systematically ?

**Comment by pxlxingliang** [comment]
Hi @QuantumMisaka 
We will try to reduce the node level of abacus.json as much as possible, so that in python it will not be too complicated to get a value.
Our current design is to divide the contents into below parts:
- general_info: some global general setting in ABACUS calculation, which is not related to the real functions
- input: the readin informations from INPUT/STRU etc. (not supported now)
- init: some initial settings by ABACUS, like k-points/k-weights etc.
- output: some outputs in or after ABACUS calculation.
- final_structure: the information of final structure, and some post process results, like energy eigenvalues, etc.

For @1041176461 mentioned information, I plan to add:
Velocity in output 
energy eigenvalues (SCF and NSCF), occupations(SCF and NSCF) in final_structure,
k-points, k-weights in init, 
dipole, atomic magnetic moment MD informations (STRU, Temperature, Pressure, Energy, Potential, Kinetic) in output

For charge density, I don't think it should be stored in abaucs.json, which has been stored in SPIN${spin}_CHG.cube. While abacus.json can show the path of those cude files.

**Comment by QuantumMisaka** [comment]
@pxlxingliang Also the running*.log have no information of 
- fixing information
- initial magnetic moment or calculated magnetic moment

which are always in STRU file, but ASE-ABACUS interfaces now cannot grab them from log file
